stackdriver_exporter.go

```diff
@@ -353,6 +355,39 @@ func main() {
 	}
 }
 
+func parseMetricTypePrefixes(inputPrefixes []string) (metricTypePrefixes []string) {
+	// only keep unique prefixes
+	uniqueKeys := make(map[string]bool)
+	uniquePrefixes := []string{}
+	for _, prefix := range inputPrefixes {
+		if _, ok := uniqueKeys[prefix]; !ok {
+			uniqueKeys[prefix] = true
+			uniquePrefixes = append(uniquePrefixes, prefix)
+		}
+	}
+
+	// drop prefixes that start with another existing prefix to avoid error:
+	// "collected metric xxx was collected before with the same name and label values"
+	slices.Sort(uniquePrefixes)
```

Comment:
This can be simplified with `slices.Compact()`.

```suggestion
	// Drop duplicate prefixes and that start with another existing prefix to avoid error:
	// "collected metric xxx was collected before with the same name and label values"
	slices.Sort(inputPrefixes)
	uniquePrefixes := slices.Compact(inputPrefixes)
```

Reply:
Done.